### ISSUE TITLE
Error in APN

### ISSUE THREAD

**Issue by andresdkm** [Edujugon/PushNotification#9]
when I try to send a push by apn I get this error
ErrorException in Apn.php line 283:
stream_socket_client(): Unable to set private key file 

I was looking and the error is due to missing

![captura de pantalla_2016-10-19_18-11-25](https://cloud.githubusercontent.com/assets/6908822/19540814/a0bd59ac-9627-11e6-8b63-41898ba7ae4e.png)

that solution is good?

**Comment by Edujugon** [comment]
@andresdkm 
Hi,

Is the issue related to sending a push notification or trying to get the APNS feedback (getFeedback method) ? or none of those methods work for you?

Aside from that, Could you check that your pem file is correctly created? maybe [this post](http://stackoverflow.com/a/26945280) helps you.

**Comment by andresdkm** [comment]
Hi

the issue is related to sending a push notification, provided the error occurred, the pem file is generated correctly, after you add the lines that I showed above works correctly.

**Comment by Edujugon** [comment]
@andresdkm 

Thanks for that bugreport.

I've already updated the code and refactored it a bit.

Please update the package version to v2.1.6.

Could you test it to make sure?

**Comment by Edujugon** [comment]
@andresdkm 
Hi,

Did you have time to try the new version? Did it work for you?

**Comment by Edujugon** [comment]
@andresdkm 
Hi, I suppose you aren't facing that issue any more. So I'm closing this..

Anyway, if you think it hasn't been solved yet or you face the same issue in the future let me know and I'll open it again.

Thanks

**Comment by fahadonline** [comment]
@Edujugon 

I am facing similar issue, my composer.json version is "edujugon/push-notification": "^2.1".

Any clue ?

{
    "message": "stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:\nerror:14094438:SSL routines:SSL3_READ_BYTES:tlsv1 alert internal error",
    "status_code": 500,
    "debug": {
        "line": 171,
        "file": "/vendor/edujugon/push-notification/src/Apn.php",
        "class": "ErrorException",
        "trace": [
            "#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'stream_socket_c...', '/opt/lampp/htdo...', 171, Array)",
            "#1/vendor/edujugon/push-notification/src/Apn.php(171): stream_socket_client('ssl://gateway.s...', 0, '', 60, 5, Resource id #357)",
            "#2 /vendor/edujugon/push-notification/src/Apn.php(212): Edujugon\\PushNotification\\Apn->openConnectionAPNS()",
            "#3 /vendor/edujugon/push-notification/src/PushNotification.php(163): Edujugon\\PushNotification\\Apn->send(Array, Array)",
            "#4 /app/Http/Controllers/API/Traits/PushNotificationTrait.php(13): Edujugon\\PushNotification\\PushNotification->send()",
            "#5 /app/Http/Controllers/API/DateController.php(112): App\\Http\\Controllers\\API\\DateController->sendPushNotification(Array, '3ce6925f43af87f...')",
            
        ]
    }
}

**Comment by Edujugon** [comment]
@fahadonline

Hi,
Was that working fine for you before updating to the last version (2.1.6) ?

**Comment by fahadonline** [comment]
@Edujugon 

Hi,

Thanks for response.
I just integrated it yesterday and tried dry run, which seemed to run and getFeedback.
But today when I executed the same code, it generated above mentioned error.

Any help in resolving it will be very helpful.

**Comment by Edujugon** [comment]
@fahadonline

That's weird because the package hasn't been updated today, it has same code as yesterday..

Today, when you get the error, are you trying same code as yesterday but changing dry_run value? Or also same dry_run value?

Regarding your second error message, that seems to be related to the cert details, no about the package..

In order to be able to reproduce your issue, could you send me the cert file and 1 token you are using? Then I'll try to replicate the issue and check it deeply..

If so, send it to me to edujugon@gmail.com

**Comment by fahadonline** [comment]
@Edujugon 

Sorry but which by "token" what are you referring to ?
I am using ".pem" certificate without any passPhrase.

**Comment by Edujugon** [comment]
I mean any device token which you are sending push notifications

**Comment by Edujugon** [comment]
@fahadonline

Have you changed the dry_run value from yesterday? Or you haven't changed anything and yesterday was working and today doesn't..

**Comment by fahadonline** [comment]
@Edujugon 

Changed no thing, same dry_run value "true"

**Comment by Edujugon** [comment]
@fahadonline

I've got your mail with the requested details :)

As soon as I have some free time (maybe tomorrow) I'll try to reproduce the issue and give you an update..

**Comment by fahadonline** [comment]
@Edujugon 

Thanks

**Comment by fahadonline** [comment]
@Edujugon 

I have resolved the issue.
I think the main problem was with using certificate without any "passPhrase".
I have got new certificate with short "passPhrase" like 123.
Though "passPhrase" is optional, not using it created problems for me.

Other than that I have used short name for ".pem" file like "apn.pem".
Also instead of copying ".pem" files to "iosCertificates" folder,
I created the new blank ".pem" files in "iosCertificates" folder, 
edited and pasted the content of downloaded ".pem" to newly created files.
Also played with "dry_run", switching it between true/false before it finally started working.
Can't remember any other weird stuff that I might have tried :)
